server/src/main/java/org/elasticsearch/index/translog/Translog.java

```diff
@@ -680,23 +719,28 @@ public Snapshot newSnapshot(long fromSeqNo, long toSeqNo) throws IOException {
      * this method will return <code>null</code>.
      */
     public Operation readOperation(Location location) throws IOException {
-        try (ReleasableLock ignored = readLock.acquire()) {
-            ensureOpen();
-            if (location.generation < getMinFileGeneration()) {
-                return null;
-            }
-            if (current.generation == location.generation) {
-                // no need to fsync here the read operation will ensure that buffers are written to disk
-                // if they are still in RAM and we are reading onto that position
-                return current.read(location);
-            } else {
-                // read backwards - it's likely we need to read on that is recent
-                for (int i = readers.size() - 1; i >= 0; i--) {
-                    TranslogReader translogReader = readers.get(i);
-                    if (translogReader.generation == location.generation) {
-                        return translogReader.read(location);
+        try {
+            readLock.lock();
+            try {
```

Comment:
Why use double try? Looks like previous version worked with single one.

Reply:
True it's more code now, but that's the price to pay here for removing some overhead :)

Reply:
Right :)
try-with-resources closes before catching exception